keps/sig-storage/3762-persistent-volume-last-phase-transition-time/README.md

```diff
@@ -0,0 +1,935 @@
+
+New e2e tests will be added for the new timestamp feature.
+
+<!--
+Based on reviewers feedback describe what additional tests need to be added prior
+implementing this enhancement to ensure the enhancements have also solid foundations.
+-->
+
+##### Unit tests
+
+<!--
+In principle every added code should have complete unit test coverage, so providing
+the exact set of tests will not bring additional value.
+However, if complete unit test coverage is not possible, explain the reason of it
+together with explanation why this is acceptable.
+-->
+
+<!--
+Additionally, for Alpha try to enumerate the core package you will be touching
+to implement this enhancement and provide the current unit coverage for those
+in the form of:
+- <package>: <date> - <current test coverage>
+The data can be easily read from:
+https://testgrid.k8s.io/sig-testing-canaries#ci-kubernetes-coverage-unit
+
+This can inform certain test coverage improvements that we want to do before
+extending the production code to implement this enhancement.
+-->
+
+Changes will be implemented in packages with sufficient unit test coverage.
+
+For any new or changed code we will add new unit tests.
+
+- `pkg/controller/volume/persistentvolume`: `2023-01-25` - `79%`
+- `pkg/apis/core/validation/`: `2023-01-25` - `82%`
+
+##### Integration tests
+
+<!--
+This question should be filled when targeting a release.
+For Alpha, describe what tests will be added to ensure proper quality of the enhancement.
+
+For Beta and GA, add links to added tests together with links to k8s-triage for those tests:
+https://storage.googleapis.com/k8s-triage/index.html
+-->
+
+This feature could be covered with integration tests only, however e2e testing provides more value and might help
+catch more bugs. Because these two kinds of tests would be almost identical, integration testing is not needed.
+
+##### e2e tests
+
+<!--
+This question should be filled when targeting a release.
+For Alpha, describe what tests will be added to ensure proper quality of the enhancement.
+
+For Beta and GA, add links to added tests together with links to k8s-triage for those tests:
+https://storage.googleapis.com/k8s-triage/index.html
+
+We expect no non-infra related flakes in the last month as a GA graduation criteria.
+-->
+
+We plan to add new e2e tests which should not interfere with any other tests, and so they could run in parallel.
+
+While the timestamp of volume phase transition will represent an accurate point in time of when it occurred, the tests
+will have to consider the time difference between user action that leads to a PV phase transition and the actual volume
+phase transition done by the PV controller.
+Based on exploratory testing we will define an appropriate time tolerance which will represent maximum time limit for
+the volume to transition phase.
```

Comment:
Won't this result in a flaky test?

Reply:
Technically yes but this kind of polling is widely used in tests already. For example [here](https://github.com/kubernetes/kubernetes/blob/5437d493da9435c9a32b244cd8bb12faf88075ae/test/e2e/e2e.go#L162). So this will not flake more than any other test that has timeout for some specific action to happen.